### PR TITLE
Add Snowflake internal build process workflow

### PR DESCRIPTION
**Pull request by VersusFacit**
resolves 38 of epic

This PR adds a workflow file for bigquery testing. It is based on https://github.com/dbt-labs/dbt-redshift/pull/732. I assume there will be some quirks to work out. Also, we'll need to add aws secrets.

### CODE REVIEW

**Review by emmyoop — CHANGES_REQUESTED** [review]
All the same questions/changes from https://github.com/dbt-labs/dbt-redshift/pull/732

File: .changes/unreleased/Features-20240318-033136.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+kind: Features
+body: ' Add new workflow for internal patch releases'
```
Comment (emmyoop) [inline .changes/unreleased/Features-20240318-033136.yaml:2]: If adapters wants these changelogs
```suggestion
body: 'Add new workflow for internal patch releases'
```

**Review by emmyoop — APPROVED** [review]
Approving as is for now with the understanding that more PRs will follow on after testing since none of this can be tested before the workflow gets merged.  Since this is a workflow that will be manually submitted there is little risk in this approach.

File: .changes/unreleased/Features-20240318-033136.yaml (hunk 1 above)
Comment (aranke) [reply .changes/unreleased/Features-20240318-033136.yaml:2]: I've been tagging GH actions changes with `Skip Changelog` (since these aren't shipped to end users).

Do we want to do that here too?

File: .github/workflows/release-internal.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        required: false
+      package_test_command:
+        description: "Package test command"
+        type: string
+        default: "python -c \"import dbt.adapters.snowflake\""
+        required: true
+      dbms_name:
+        description: "The name of the warehouse the adapter connects to."
+        type: string
+        default: "snowflake"
```
Comment (aranke) [inline .github/workflows/release-internal.yml:22]: Would this ever be a value besides `snowflake`?

File: .github/workflows/release-internal.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        type: string
+        required: true
+      sha:
+        description: "The sha to use (leave empty to use latest on main)"
+        type: string
+        required: false
+      package_test_command:
+        description: "Package test command"
+        type: string
+        default: "python -c \"import dbt.adapters.snowflake\""
```
Comment (aranke) [inline .github/workflows/release-internal.yml:37]: Nit; should we change this to `import dbt.adapters as adapters; assert hasattr(adapters, 'snowflake')` to make this a bit clearer?

File: .github/workflows/release-internal.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    shell: bash
+
+env:
+  PYTHON_TARGET_VERSION: 3.11
+
+jobs:
+  invoke-reusable-workflow:
+    name: Build and Release Internally
+
+    uses: VersusFacit/dbt-release/.github/workflows/internal-archive-release.yml@main
```
Comment (aranke) [inline .github/workflows/release-internal.yml:56]: Should we migrate this workflow to the `dbt-labs` namespace before merging?

**Review by aranke — COMMENTED** [review]
Mostly LGTM, few nits
